File: csharp/ZeroBuffer.Tests/TestDataPatterns.cs
using System;

namespace ZeroBuffer.Tests
{
    /// <summary>
    /// Shared test data patterns for consistent data generation across processes
    /// </summary>
    public static class TestDataPatterns
    {
        /// <summary>
        /// Generate test data for a frame based on size and sequence number
        /// </summary>
        public static byte[] GenerateFrameData(int size, ulong sequence)
        {
            var data = new byte[size];
            for (int i = 0; i < size; i++)
            {
                data[i] = (byte)(((ulong)i + sequence) % 256);
            }
            return data;
        }

        /// <summary>
        /// Generate simple test data for a frame based only on size
        /// Used when sequence number is not known at write time
        /// </summary>
        public static byte[] GenerateSimpleFrameData(int size)
        {
            var data = new byte[size];
            for (int i = 0; i < size; i++)
            {
                data[i] = (byte)(i % 256);
            }
            return data;
        }

        /// <summary>
        /// Verify that frame data matches the simple pattern
        /// </summary>
        public static bool VerifySimpleFrameData(byte[] data)
        {
            for (int i = 0; i < data.Length; i++)
            {
                if (data[i] != (byte)(i % 256))
                    return false;
            }
            return true;
        }

        /// <summary>
        /// Generate test metadata based on size
        /// </summary>
        public static byte[] GenerateMetadata(int size)
        {
            var metadata = new byte[size];
            for (int i = 0; i < size; i++)
            {
                metadata[i] = (byte)(i % 256);
            }
            return metadata;
        }
    }
}

File: cpp/step_definitions/test_data_patterns.cpp
#include "test_data_patterns.h"

namespace zerobuffer {
namespace steps {

std::vector<uint8_t> TestDataPatterns::generateFrameData(size_t size, uint64_t sequence) {
    std::vector<uint8_t> data(size);
    for (size_t i = 0; i < size; ++i) {
        data[i] = static_cast<uint8_t>((i + sequence) % 256);
    }
    return data;
}

std::vector<uint8_t> TestDataPatterns::generateSimpleFrameData(size_t size) {
    std::vector<uint8_t> data(size);
    for (size_t i = 0; i < size; ++i) {
        data[i] = static_cast<uint8_t>(i % 256);
    }
    return data;
}

bool TestDataPatterns::verifySimpleFrameData(const uint8_t* data, size_t size) {
    for (size_t i = 0; i < size; ++i) {
        if (data[i] != static_cast<uint8_t>(i % 256)) {
            return false;
        }
    }
    return true;
}

bool TestDataPatterns::verifySimpleFrameData(const std::vector<uint8_t>& data) {
    return verifySimpleFrameData(data.data(), data.size());
}

std::vector<uint8_t> TestDataPatterns::generateMetadata(size_t size) {
    std::vector<uint8_t> metadata(size);
    for (size_t i = 0; i < size; ++i) {
        metadata[i] = static_cast<uint8_t>(i % 256);
    }
    return metadata;
}

} // namespace steps
} // namespace zerobuffer